engine/classes/Elgg/Di/DiContainer.php

```diff
@@ -133,6 +131,9 @@ public function setFactory($name, $callable, $shared = true) {
 	 * @throws \InvalidArgumentException
 	 */
 	public function setClassName($name, $class_name, $shared = true) {
+		if (0 === strpos($name, '__')) {
```

Comment:
I know it's overkill, but wouldn't `$str->startsWith('__')` be more readable? It kind of annoys me that strings aren't objects in PHP.